### PR TITLE
fix lockheld checks, throw meaningful logic errors rather than abort()

### PR DESCRIPTION
**Pull request by Greg-Griffith**

### CODE REVIEW

File: src/deadlock-detection/threaddeadlock.cpp [hunk 1]
```diff
@@ -480,9 +480,10 @@ void AssertLockNotHeldInternal(const char *pszName, const char *pszFile, unsigne
         {
             if (entry.first == cs)
             {
+                // this thread is holding the lock when it should not be
                 fprintf(stderr, "Assertion failed: lock %s held in %s:%i; locks held:\n%s", pszName, pszFile, nLine,
                     _LocksHeld().c_str());
-                abort();
+                throw std::logic_error("AssertLocNotkHeld failure");
```
Comment (ptschip) [inline src/deadlock-detection/threaddeadlock.cpp]: -AssertLocNotHeld
+AssertLockNotHeld

Comment (Greg-Griffith) [reply src/deadlock-detection/threaddeadlock.cpp]: fixed, thanks

**Review by ptschip — APPROVED** [review]

File: src/deadlock-detection/threaddeadlock.cpp [hunk 2]
```diff
@@ -480,9 +480,10 @@ void AssertLockNotHeldInternal(const char *pszName, const char *pszFile, unsigne
         {
             if (entry.first == cs)
             {
+                // this thread is holding the lock when it should not be
                 fprintf(stderr, "Assertion failed: lock %s held in %s:%i; locks held:\n%s", pszName, pszFile, nLine,
                     _LocksHeld().c_str());
-                abort();
+                throw std::logic_error("AssertLockNotkHeld failure");
```
Comment (gandrewstone) [inline src/deadlock-detection/threaddeadlock.cpp]: "AssertLockNotkHeld failure" -> "AssertLockNotHeld failure"

Comment (Greg-Griffith) [reply src/deadlock-detection/threaddeadlock.cpp]: fixed

**Pull request merged**